client/src/document/code/copy.ts

```diff
@@ -0,0 +1,79 @@
+export function addCopyToClipboardButton(
```

Comment:
I'm assuming there's no change to the functions copied over from `hooks.ts` for ease of reviewing.